classes/controller/AdminController.php

```diff
@@ -4789,4 +4798,74 @@ protected function isAnonymousAllowed()
     {
         return $this->allowAnonymous;
     }
+
+    /**
+     * Get the buttons provided by hooks and merge them into the already defined page toolbar buttons
+     *
+     * @throws TypeException
+     */
+    protected function mergeExtraToolbarButtons(): void
+    {
+        $toolbarButtonsCollection = new ActionsBarButtonsCollection();
+
+        // Get previously assigned toolbar buttons of the controller
+        $controllerButtons = $this->page_header_toolbar_btn;
+        if (!empty($controllerButtons)) {
+            // Build ActionsBarButton based on array setting from the controller and add it to collection
+            foreach ($controllerButtons as $controllerButton) {
+                $toolbarButtonsCollection->add(
+                    new ActionsBarButton(
+                        $controllerButton['class'] ?? '',
+                        $controllerButton,
+                        $controllerButton['desc'] ?? ''
+                    )
+                );
+            }
+        }
+
+        // Get the toolbar buttons defined by hooks and add them to the main collection
+        try {
```

Comment:
I didn't know about this way to execute hooks! From what I understood from the class it is used for hooks that return values and to assert the returned content matches expected classes

I wonder if this is required in this case though? Because a simple `Hook::exec` would work as well since the collection is passed by reference So any modification to the collection in the hook will be done on the same referenced object anyay so no need to return it in the hook function